state-chain/pallets/cf-swapping/src/tests.rs

```diff
@@ -1771,3 +1772,98 @@ mod private_channels {
 		});
 	}
 }
+
+#[cfg(test)]
+mod affiliates {
+
+	use super::*;
+
+	use cf_traits::mocks::account_role_registry::MockAccountRoleRegistry;
+
+	#[test]
+	fn register_address_and_request_withdrawal_success() {
+		new_test_ext().execute_with(|| {
+			const SHORT_ID: AffiliateShortId = AffiliateShortId(0);
+			const BALANCE: AssetAmount = 200;
+			let withdrawal_address: EncodedAddress = EncodedAddress::Eth(Default::default());
+
+			assert_ok!(Swapping::register_affiliate(
+				OriginTrait::signed(BROKER),
+				SHORT_ID,
+				withdrawal_address.clone()
+			));
+
+			let affiliate_account_id = AffiliateIdMapping::<Test>::get(BROKER, SHORT_ID)
+				.expect("Affiliate must be registered!");
+
+			MockBalance::credit_account(&affiliate_account_id, Asset::Usdc, BALANCE);
+
+			assert_ok!(Swapping::affiliate_withdrawal_request(
+				OriginTrait::signed(BROKER),
+				affiliate_account_id
+			));
+
+			assert_event_sequence!(
+				Test,
+				RuntimeEvent::Swapping(Event::AffiliateRegistration { .. }),
+				RuntimeEvent::Swapping(Event::WithdrawalRequested { .. }),
+			);
+		});
```

Comment:
I think in addition to checking events, we should also check that the state has actually changed (the affiliate's balance and maybe the fact that the egress API was called depending if not too difficult).